mllib/src/main/scala/org/apache/spark/mllib/evaluation/MultilabelMetrics.scala

```diff
@@ -178,7 +178,11 @@ private[evaluation] class MultilabelSummarizer extends Serializable {
    * @return This MultilabelSummarizer object.
    */
   def add(predictions: Array[Double], labels: Array[Double]): this.type = {
-    val intersection = predictions.intersect(labels)
+    val intersection = if (predictions.isEmpty || labels.isEmpty) {
```

Comment:
In Scala 2.13.5
```
Welcome to Scala 2.13.5 (OpenJDK 64-Bit Server VM, Java 1.8.0_232).
Type in expressions for evaluation. Or try :help.

scala> Array.empty[Double].intersect(Array(0.0))
val res0: Array[Double] = Array()

scala> Array(0.0).intersect(Array.empty[Double])
val res1: Array[Double] = Array()
```

but in Scala 2.13.6

```
Welcome to Scala 2.13.6 (OpenJDK 64-Bit Server VM, Java 1.8.0_232).
Type in expressions for evaluation. Or try :help.

scala> Array.empty[Double].intersect(Array(0.0))
java.lang.ClassCastException: [Ljava.lang.Object; cannot be cast to [D
  ... 32 elided

scala> Array(0.0).intersect(Array.empty[Double])
java.lang.ClassCastException: [Ljava.lang.Object; cannot be cast to [D
  ... 32 elided
```

`Array()` is replaced by throw `ClassCastException`, this change is uncomfortable ... @dongjoon-hyun @maropu do you think this is a bug of Scala 2.13.6?

Reply:
I submitted an issue to consult the Scala community  [https://github.com/scala/bug/issues/12403](https://github.com/scala/bug/issues/12403)

Reply:
Thank you for filing the upstream issue. Ya, it looks like a breaking behavior change.

Reply:
As @LuciferYang mentioned at the Scala community, I also confirmed that only Scala 2.13.6 has this bug unlike Scala 2.13.5/3.0.0.

```
scala3-3.0.0:$ bin/scala
scala> Array.empty[Double].intersect(Array(0.0))
val res0: Array[Double] = Array()

scala-2.13.6:$ bin/scala
Welcome to Scala 2.13.6 (OpenJDK 64-Bit Server VM, Java 1.8.0_292).
Type in expressions for evaluation. Or try :help.

scala> Array.empty[Double].intersect(Array(0.0))
java.lang.ClassCastException: [Ljava.lang.Object; cannot be cast to [D
  ... 32 elided
```

Reply:
In this case, we had better skip Scala 2.13.6.

Reply:
@dongjoon-hyun agree with you. we'd better skip this version